nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,220 @@
+   *     to false.
+   * @param path Directory where all files generated/read by Fix Serialization package resides.
+   */
+  public FixSerializationConfig(boolean autofixEnabled, String path) {
+    // if autofixEnabled is false, all flags will be false regardless of their given value in json
+    // config file.
+    Preconditions.checkNotNull(path);
+
+    Document document = null;
+    if (autofixEnabled) {
+      try {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        document = builder.parse(Files.newInputStream(Paths.get(path)));
+        document.normalize();
+      } catch (Exception e) {
+        throw new RuntimeException("Error in reading/parsing config at path: " + path + "\n" + e);
+      }
+    }
+    this.outputDirectory =
+        XMLUtil.getValueFromTag(document, "serialization:path", String.class).orElse(null);
+    Preconditions.checkNotNull(
+        this.outputDirectory, "Error in FixSerialization Config: Output path cannot be null");
+    suggestEnabled =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "active", Boolean.class)
+                .orElse(false)
+            && autofixEnabled;
+    suggestEnclosing =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "enclosing", Boolean.class)
+                .orElse(false)
+            && suggestEnabled;
+    logErrorEnabled =
+        XMLUtil.getValueFromAttribute(document, "serialization:error", "active", Boolean.class)
+                .orElse(false)
+            && autofixEnabled;
+    logErrorEnclosing =
+        XMLUtil.getValueFromAttribute(document, "serialization:error", "enclosing", Boolean.class)
+                .orElse(false)
+            && logErrorEnabled;
+    String nullableAnnot =
+        XMLUtil.getValueFromTag(document, "serialization:annotation:nullable", String.class)
+            .orElse("javax.annotation.Nullable");
+    String nonnullAnnot =
+        XMLUtil.getValueFromTag(document, "serialization:annotation:nonnull", String.class)
+            .orElse("javax.annotation.Nonnull");
+    this.annotationFactory = new AnnotationFactory(nullableAnnot, nonnullAnnot);
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Validates whether {@code suggestEnabled} is active and the element involved in the error is
+   * accessible and not received in bytecodes.
+   *
+   * @param trees Trees object.
+   * @param symbol Symbol of the targeted element to perform type change.
+   * @return true, if the element is accessible and {@code suggestEnabled} is active.
+   */
+  public boolean canFixElement(Trees trees, Element symbol) {
+    if (!suggestEnabled) return false;
+    if (trees == null || symbol == null) return false;
+    return trees.getPath(symbol) != null;
+  }
+
+  /** Builder class for Config */
+  public static class FixSerializationConfigBuilder {
```

Comment:
This class is used by a lot in ```AutoFixer``` since it runs ```NullAway``` with different modes. I created this class here so we could reflect ```FixSerializationConfig``` changes to it's builder as well. Would you like me remove it ?